### ISSUE TITLE
[Feature Request] Optimizer parameters schedulers

### ISSUE THREAD

**Issue by thiagocrepaldi** [microsoft/onnxruntime#5165]
**Is your feature request related to a problem? Please describe.**
Raised on the same discussion as https://github.com/microsoft/onnxruntime/issues/5164, where internal projects need to change hyper parameters between training steps

**System information**
- latest Optimized API

**Describe the solution you'd like**
An optimizer scheduler in which users could use methods as hyper parameter values, giving the ability to the method return different values over the training course

**Additional context**
ONNX Runtime already supports LR schedulers and Loss scaling, but not other optimizer parameters
